src/shared/ui/survey/GeolocationItem/GeolocationItem.tsx

```diff
@@ -0,0 +1,119 @@
+import { useEffect, useMemo, FC } from 'react';
+import { useState } from 'react';
+
+import NativeGeolocation from '@react-native-community/geolocation';
+import { Button } from '@tamagui/button';
+import { styled } from '@tamagui/core';
+import { useTranslation } from 'react-i18next';
+import Permissions, { RESULTS } from 'react-native-permissions';
+
+import { IS_ANDROID, LOCATION_PERMISSIONS } from '@app/shared/lib';
+
+import { Center, GeolocationIcon, Text } from '../..';
+
+const GeolocationButton = styled(Button, {
+  backgroundColor: '$blue',
+  borderRadius: 0,
+  color: '$white',
+  fontWeight: '900',
+  minWidth: '10%',
+});
+
+type Coordinates = {
+  latitude: number;
+  longitude: number;
+};
+
+type Props = {
+  onChange: (value: Coordinates) => void;
+  value: Coordinates | null;
+};
+
+const GeolocationItem: FC<Props> = ({ onChange, value = null }) => {
+  const { t } = useTranslation();
+  const [coordinates, setCoordinates] = useState<Coordinates | null>(value);
```

Comment:
The component should be a dumb component. Hence, it should depend the the `value`